src/Dictionaries/LibraryDictionarySource.h

```diff
@@ -68,18 +69,33 @@ class LibraryDictionarySource final : public IDictionarySource
     std::string toString() const override;
 
 private:
-    Poco::Logger * log;
+    String getDictAttributesString();
+
+    static String getDictIdsString(const std::vector<UInt64> & ids);
+
+    static String getLibrarySettingsString(const Poco::Util::AbstractConfiguration & config, const std::string & config_root);
 
-    LocalDateTime getLastModification() const;
+    Poco::Logger * log;
 
     const DictionaryStructure dict_struct;
     const std::string config_prefix;
     const std::string path;
+    const std::string dictionary_id;
+
     Block sample_block;
-    SharedLibraryPtr library;
+    Context context;
+
+    LibraryBridgeHelperPtr bridge_helper;
     ExternalResultDescription description;
-    std::shared_ptr<CStringsHolder> settings;
-    void * lib_data = nullptr;
+
+    String createDictID()
```

Comment:
Why not just use UUID https://github.com/ClickHouse/ClickHouse/blob/master/src/Core/UUID.h#L1-L25?